### PR TITLE
[pmdk][libpmemobj-cpp] Updates and improvements

### LINKED ISSUE #4683: [New port request] libpmemobj-cpp, pmdk-convert
Please update PMDK to version 1.5

We would also like two more packages to be added:
- libpmemobj-cpp: https://github.com/pmem/libpmemobj-cpp
- pmdk-convert: https://github.com/pmem/pmdk-convert

### PR DESCRIPTION
**Pull request by ras0219-msft**
Fixes #4683

### CODE REVIEW

File: ports/libpmemobj-cpp/disable-perl.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
+ 
+-if(NOT PERL_FOUND)
++if(NOT PERL_FOUND AND DEVELOPER_MODE)
+ 	message(FATAL_ERROR "Perl not found")
+ endif()
+-if (PERL_VERSION_STRING VERSION_LESS 5.16)
++if (PERL_VERSION_STRING VERSION_LESS 5.16 AND DEVELOPER_MODE)
+ 	message(FATAL_ERROR "Too old Perl (<5.16)")
+ endif()
```
Comment (marcinslusarz) [inline ports/libpmemobj-cpp/disable-perl.patch]: Please submit this upstream.

File: ports/pmdk/portfile.cmake [hunk 2]
```diff
@@ -16,51 +17,37 @@ vcpkg_from_github(
     SHA512 f66e4edf1937d51abfa7c087b65a64109cd3d2a8d9587d6c4fc28a1003d67ec1f35a0011c9a9d0bfe76ad7227be83e86582f8405c988eac828d8ae5d0a399483
     HEAD_REF master
     PATCHES
-        "${CMAKE_CURRENT_LIST_DIR}/addPowerShellExecutionPolicy.patch"
-        "${CMAKE_CURRENT_LIST_DIR}/v141.patch"
+        addPowerShellExecutionPolicy.patch
+        v141.patch
```
Comment (marcinslusarz) [inline ports/pmdk/portfile.cmake]: I think those patches are no longer applicable.

**Comment by JackBoosY** [comment]
Since libpmemobj-cpp and pmdk have been updated to a newer version, do we still need this PR?

**Comment by PhoebeHui** [comment]
Currently, libpmemobj-cpp upgrade to 1.8, and pmdk upgrade to 1.7. @ras0219-msft , could we close this PR?

**Pull request closed without merging**